courseware/api.py

```diff
@@ -96,11 +138,18 @@ def is_existing_edx_user(user):
         }
     )
     response = req_session.get(
-        edx_url(f"{OPENEDX_USER_ACCOUNT_DETAIL_PATH}/{user.username}")
+        edx_url(f"{OPENEDX_USER_ACCOUNT_DETAIL_PATH}"), params={"email": user.email}
     )
     # user already exists on edx
     if response.status_code == status.HTTP_200_OK and is_json_response(response):
-        return response.json().get("email") == user.email
+        users = response.json()
```

Comment:
I don't think a method named `is_existing_edx_user` should be side-effecting like this. Maybe it makes more sense to have this be a `get_existing_openedx_user` (name it openedx, not edx, since it's not edx.org), which would return an object wrapping both `user` and the user data from the API response. Then on that wrapper class you can have properties that determine whether things on the xpro side need an update.

I'm thinking something like this:

```python
from dataclasses import dataclass

@dataclass(frozen=True)
class OpenEdxUser:
    user: User
    openedx_data: dict

    def is_username_match(self):
        return self.user.username == self.opnedx_data["username"]

user = OpenEdxUser(user, response.json()[0]
```

Then where this method is called, check this and separately make the repair in the db.